extensions/kafka-client/runtime/src/main/java/io/quarkus/kafka/client/runtime/KafkaAdminClient.java

```diff
@@ -61,23 +61,23 @@ public Collection<ConsumerGroupDescription> getConsumerGroups() throws Interrupt
                 .values();
     }
 
-    public boolean deleteTopic(String name) {
+    public boolean deleteTopic(final String name) {
         Collection<String> topics = new ArrayList<>();
         topics.add(name);
         DeleteTopicsResult dtr = client.deleteTopics(topics);
         return dtr.topicNameValues() != null;
     }
 
-    public boolean createTopic(KafkaCreateTopicRequest kafkaCreateTopicRq) {
+    public boolean createTopic(final KafkaCreateTopicRequest kafkaCreateTopicRq) {
         var partitions = Optional.ofNullable(kafkaCreateTopicRq.getPartitions()).orElse(1);
```

Comment:
I agree with the choice, but we need to be clear about it.
If you use a dev service, the broker is configured to create a single partition if you do not configure the number of partitions.

However, if you use your own Kafka, the default number of partitions can be different. Unfortunately, I don't believe we can know this magic number.

Reply:
@ozangunalp WDYT?

Reply:
A good example is our Confluent instance that holds an unchanged default value of 3 as default for Kafka.
Using this method whitout settings parameters doesn't work straight away.

By choice you mean the `.orElse(1)` part? This was original to the code and not changed by me :)

Reply:
In theory, we can tap into the broker configs to extract values for `num.partitions` and `default.replication.factor`.

Reply:
We can use `.orElse(1)` but the UI should contain a message.